### PR TITLE
[core] Convert operator supports `u2, u3, u6` types

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Add new types `u2, u3, u6` to Convert operator.
 - Expand `element::Iterator` to support NF4 and `BitProxy` to support NF4 conversion.
 - Update tensor to calculate correctly byte size for `u3, u6` types.
 - Fix NF4 <-> conversion to always use byte pack/unpack and quantization when convert to/from floating point. In future the conversion for NF4 will be limited to f32 -> NF4.

### Tickets:
 - [CVS-127000](https://jira.devtools.intel.com/browse/CVS-127000)
 - [CVS-128024](https://jira.devtools.intel.com/browse/CVS-128024)

### CODE REVIEW

**Comment by praasz** [comment]
Blocked by PR #23279 .

Last 5 commits impalements new feature in Convert operator.

**Comment by mlukasze** [comment]
> Blocked by PR #23279 .

blocker has been merged

File: src/core/src/op/convert.cpp [hunk 1]
```diff
@@ -15,50 +15,31 @@ namespace ov {
...
-        template <element::Type_t ET,
-                  class T,
-                  class T_ET,
-                  class U = ov::fundamental_type_for<ET>,
-                  typename std::enable_if<!is_lp_type(ET) && !IS_ARG_ET_LP>::type* = nullptr>
-        static result_type visit(const T* arg, Tensor& out, const size_t count, T_ET&&) {
-            reference::convert(arg, out.data<U>(), count);
+        template <element::Type_t ET_OUT, class InputIter, class TO = ov::fundamental_type_for<ET_OUT>>
+        static result_type visit(InputIter arg, Tensor& out, const size_t count) {
+            reference::convert(arg, element::iterator<ET_OUT>(out.data()), count);
```
Comment (mitruska) [inline src/core/src/op/convert.cpp:42]: Could `reinterpret_cast` be needed accordingly to the cast for the input data?
```suggestion
        static result_type visit(InputIter arg, Tensor& out, const size_t count) {
            reference::convert(arg, element::iterator<ET_OUT>(reinterpret_cast<TO*>(out.data())), count);
```

**Review by mitruska — APPROVED** [review]

File: src/core/src/op/convert.cpp (hunk 1 above)
Comment (praasz) [reply src/core/src/op/convert.cpp:42]: No is not required here.

**Pull request merged**